test/unittests/test_cpp.cpp

```diff
@@ -120,3 +124,35 @@ TEST(cpp, host_call)
 
     example_host_destroy_context(host_context);
 }
+
+TEST(cpp, result_raii)
+{
+    static auto release_called = false;
+    release_called = false;
+    auto release_fn = [](const evmc_result*) noexcept { release_called = true; };
+
+    {
+        auto raw_result = evmc_result{};
+        raw_result.status_code = EVMC_INTERNAL_ERROR;
+        raw_result.release = release_fn;
+
+        auto raii_result = evmc::result{raw_result};
+        raii_result.gas_left = -1;
+
+        auto raw_result2 = raii_result.raw();
+        EXPECT_EQ(raw_result2.status_code, EVMC_INTERNAL_ERROR);
+        EXPECT_EQ(raw_result2.gas_left, -1);
+        EXPECT_EQ(raw_result.release, release_fn);
```

Comment:
Should `result2` have `release_fn` and `result` have `nullptr`? Can you check for both here since they are ought to differ?

Reply:
Both will have the same `release` function. The check added.